File: 프로그래머스/1/42840. 모의고사/모의고사.js
function solution(answers) {
    let answer = new Array();
    
    const pattern1 = [1, 2, 3, 4, 5];
    const pattern2 = [2, 1, 2, 3, 2, 4, 2, 5];
    const pattern3 = [3, 3, 1, 1, 2, 2, 4, 4, 5, 5];
    
    let stu1 = 0;
    let stu2 = 0;
    let stu3 = 0;
    for(let i = 0; i < answers.length; i++) {
        const x = answers[i];
        
        if(pattern1[i % 5] == x) stu1++;
        if(pattern2[i % 8] == x) stu2++;
        if(pattern3[i % 10] == x) stu3++;
    }
    
    const max = Math.max(stu1, Math.max(stu2, stu3));
    if(max === stu1) answer.push(1);
    if(max === stu2) answer.push(2);
    if(max === stu3) answer.push(3);
    
    return answer;
}

File: 프로그래머스/1/42840. 모의고사/모의고사.java
import java.util.*;

class Solution {
    
    private static int stu1, stu2, stu3;
    private static int[] pattern1 = {1, 2, 3, 4, 5};
    private static int[] pattern2 = {2, 1, 2, 3, 2, 4, 2, 5};
    private static int[] pattern3 = {3, 3, 1, 1, 2, 2, 4, 4, 5, 5};
    
    public List<Integer> solution(int[] answers) {
        List<Integer> list = new ArrayList<>();
        
        for(int i = 0; i < answers.length; i++) {
            int answer = answers[i];
            
            if(pattern1[i % 5] == answer) stu1++;
            if(pattern2[i % 8] == answer) stu2++;
            if(pattern3[i % 10] == answer) stu3++;
        }
        
        int max = Math.max(stu1, Math.max(stu2, stu3));
        if(stu1 == max) list.add(1);
        if(stu2 == max) list.add(2);
        if(stu3 == max) list.add(3);
        
        return list;
    }
}